tests/RobotFramework/tests/pkcs11/private_key_storage.robot

```diff
@@ -31,6 +31,63 @@ Use Private Key in SoftHSM2 using tedge-p11-server
 
     Execute Command    tedge reconnect c8y
 
+Select Private key using PKCS#11 URI
+    [Documentation]    Make sure that we can select different keys and tokens using a PKCS#11 URI.
+    ...    The URI can either point to a specific key, or to a specific token where we will attempt to find a key.
+    ...
+    ...    To ensure that correct key is selected and reduce the need to generate and upload different keys and
+    ...    certificates, we'll only be using one key and we'll only import it to the chosen token, keeping other tokens
+    ...    empty.
+
+    # configure tedge
+    Execute Command    tedge config set c8y.url "$(echo ${C8Y_CONFIG.host} | sed 's|https?://||g')"
+    Execute Command    tedge config set mqtt.bridge.built_in true
+    Execute Command    tedge config set device.cryptoki.mode socket
+
+    Set tedge-p11-server Configuration    pkcs11_url=
+    Execute Command    sudo -u tedge /usr/bin/init_softhsm.sh --self-signed --device-id "${DEVICE_SN}" --pin 123456
+
+    # Upload the self-signed certificate
+    Execute Command
+    ...    cmd=sudo env C8Y_USER='${C8Y_CONFIG.username}' C8Y_PASSWORD='${C8Y_CONFIG.password}' tedge cert upload c8y
+
+    Execute Command    tedge reconnect c8y
+
+    # expect failure if we try to use a token that doesnt exist
+    Set tedge-p11-server Configuration    pkcs11_url=pkcs11:token=asdf
+    Restart Service    tedge-p11-server
+    Execute Command    tedge reconnect c8y    exp_exit_code=!0
```

Comment:
There are so many reason for `tedge reconnect` to fail that I would assess that we really get the error we expect.

```suggestion
    ${stderr}=    Execute Command    tedge reconnect c8y    exp_exit_code=!0    stdout=false    stderr=true
    Should Contain    ${stderr}    tedge_p11_server
    Should Contain    ${stderr}    Failed to find a signing key: Didn't find a slot to use
```

Reply:
fixed in a8a732e. The suite was also refactored a bit in the next commit so it should be a bit easier to read.